datafusion/sql/tests/integration_test.rs

```diff
@@ -3252,6 +3252,17 @@ fn test_select_distinct_order_by() {
     assert_eq!(err.to_string(), expected);
 }
 
+#[test]
+fn select_order_by_with_cast() {
```

Comment:
I verified this test fails without the change in this PR 👍 

---- select_order_by_with_cast stdout ----
thread 'select_order_by_with_cast' panicked at 'assertion failed: `(left == right)`
  left: `"Sort: first_name AS first_name ASC NULLS LAST\n  Projection: first_name AS first_name\n    Projection: person.first_name AS first_name\n      TableScan: person"`,
 right: `"Sort: CAST(first_name AS first_name AS Int32) ASC NULLS LAST\n  Projection: first_name AS first_name\n    Projection: person.first_name AS first_name\n      TableScan: person"`', datafusion/sql/tests/integration_test.rs:2433:5